clients/clients.go

```diff
@@ -299,15 +310,14 @@ func (c *Clients) onBotOptionsEvent(client *gomatrix.Client, event *gomatrix.Eve
 	}
 }
 
-func (c *Clients) onRoomMemberEvent(client *gomatrix.Client, event *gomatrix.Event) {
-	if *event.StateKey != client.UserID {
-		return // not our member event
-	}
-	m := event.Content["membership"]
-	membership, ok := m.(string)
-	if !ok {
+func (c *Clients) onRoomMemberEvent(client *mautrix.Client, event *mevt.Event) {
+	if err := event.Content.ParseRaw(mevt.StateMember); err != nil {
 		return
 	}
+	if *event.StateKey != client.UserID.String() {
```

Comment:
While I didn't write this line, I understand from https://matrix.org/docs/spec/client_server/r0.6.1#state-event-fields that a state key should always exist. I'll add a nil check to be sure if you think that it could be nil.

Reply:
It can be `nil`, nothing stops anyone from sending an `m.room.member` event without a state key.

Reply:
done :) Also added a similar nil check to `onBotOptionsEvent`.